gobblin-metastore/src/main/java/gobblin/metastore/FsStateStore.java

```diff
@@ -174,11 +172,13 @@ public void putAll(String storeName, String tableName, Collection<T> states)
     } finally {
       closer.close();
     }
+
+    Path tablePath = new Path(new Path(this.storeRootDir, storeName), tableName);
+    HadoopUtils.renamePath(this.fs, tmpTablePath, tablePath);
```

Comment:
As long as it is atomic and doesn't create a partial file at `tablePath`. Since this is renaming a single file, it should be fine.